Cython/Compiler/ExprNodes.py

```diff
@@ -9532,11 +9526,12 @@ def calculate_constant_result(self):
         self.constant_result = (
             self.key.constant_result, self.value.constant_result)
 
-    def analyse_types(self, env):
+    def analyse_types(self, is_dict, env):
```

Comment:
> `analyse_types()` is a base class API method. You cannot just change its signature arbitrarily inside of the syntax tree. It's ok to unpack coerced values later, if we decide that there is a better way to process them. That's better than distributing full global knowledge everywhere first.

The problem is the values are IntNodes, and instead of getting wrapped, they just get `type` changed.

Reply:
I suppose we could fix this by just using CoerceToPyTypeNode directly.

Reply:
Should be fixed now.